### PR TITLE
Update to `cardano-api-8.23.1.0`

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Update to `cardano-api-8.23.1.0`
# uncomment types applicable to the change:
  type:
   - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
   - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Requires to build a new release of cadano-api which includes:
- https://github.com/input-output-hk/cardano-api/pull/287
for additional ledger reexports and `AnyEraInEon` changes.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/cardano-cli.cabal [hunk 1]
```diff
@@ -200,7 +200,7 @@ library
                       , binary
                       , bytestring
                       , canonical-json
-                      , cardano-api ^>= 8.22
```
Comment (carbolymer) [inline cardano-cli/cardano-cli.cabal:203]: TODO:
- [ ] use 8.24 after release
- [ ] bump flake.lock
- [ ] bump chap index in cabal.project

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Actions.hs [hunk 2]
```diff
@@ -277,10 +277,44 @@ dpGovActionProtocolParametersUpdate = \case
...
+  <*> pDRepPoolThd
+  <*> pMinCommitteeSize
+  <*> pMinCommitteeTermLength
+  <*> pMinGovActionLifeTime
+  <*> (SJust . toShelleyLovelace <$> pGovActionDeposit)
+  <*> pDRepDeposit
+  <*> ppDRepActivity
+
+ppDRepActivity :: Parser (StrictMaybe EpochNo)
+ppDRepActivity = pure SNothing -- TODO Conway
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Actions.hs]: TODO Parsers are stubbed to make changes smaller. To be implemented in the next PR.

**Comment by carbolymer** [comment]
Build fails currently because it requires unreleased cardano-api changes. To be fixed prior to merge.

**Review by Jimbo4350 — DISMISSED** [review]
I'm not sold on `AnyEraInEon`

**Comment by newhoggy** [comment]
As agreed, `AnyEraInEon` will be renamed to `EraInEon` in a future PR.

**Review by newhoggy — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**